mlir/test/lib/Dialect/Tensor/TestTensorTransforms.cpp

```diff
@@ -77,6 +77,10 @@ struct TestTensorTransforms
       llvm::cl::desc("Test folding ops into tensor.pack and tensor.unpack"),
       llvm::cl::init(false)};
 
+  Option<bool> testDecomposeConcat{
```

Comment:
Can we plug this into a simple TD ApplyPattern rather than add more to this one-off C++ test ?
It is both shorther and more localized.

Reply:
Sounds good to me! I just wasn't sure if we were happy with having testing stories being tied to transform ops.

Reply:
In general, I think we should move to transform ops for everything that requires such custom C++ test logic and make sense as a transform op.
C++ tests that are modeled like this one:
  1. are throwaway and never used outside of test, at least transforms are reusable.
  2. are opaque and hidden behind a name, at least transforms make the action of the test very clear.
  3. have intertwined `runOnOperation` implementation that just keeps on growing